src/bun.js/node/node_fs.zig

```diff
@@ -5216,7 +5222,8 @@ pub const NodeFS = struct {
             switch (comptime ExpectedType) {
                 Dirent => {
                     entries.append(.{
-                        .name = bun.String.createUTF8(name_to_copy),
+                        .name = bun.String.createUTF8(utf8_name),
+                        .path = bun.String.createUTF8(std.fs.path.dirname(bun.path.join(&[_]string{ root_basename, name_to_copy }, .auto)).?),
```

Comment:
is `.?` guaranteed? What if you did `readdir("/")`?

Reply:
good point, I should use `bun.path.dirname` that handles that